src/main/java/org/broadinstitute/hellbender/tools/spark/sv/evidence/BreakpointEvidence.java

```diff
@@ -410,25 +518,60 @@ public SplitRead( final GATKRead read, final ReadMetadata metadata, final boolea
 
         private SplitRead( final Kryo kryo, final Input input ) {
             super(kryo, input);
-            cigar = input.readString();
             tagSA = input.readString();
-            primaryAlignmentForwardStrand = input.readBoolean();
             primaryAlignmentClippedAtStart = input.readBoolean();
             saMappings = parseSATag(tagSA);
         }
 
+        /**
+         * Constructor used by fromStringRep() to construct BreakpointEvidence
+         */
+        private SplitRead(final SVInterval interval, final int weight,
+                          final String templateName, final TemplateFragmentOrdinal fragmentOrdinal,
+                          final boolean validated, final boolean forwardStrand,
+                          final String cigarString, final int mappingQuality,
+                          final int templateSize, final String readGroup,
+                          final List<StrandedInterval> distalTargets,
+                          final ReadMetadata readMetadata) {
+            super(interval, weight, templateName, fragmentOrdinal, validated, forwardStrand, cigarString,
+                    mappingQuality, templateSize, readGroup);
+            // NOTE: can't identically reconstruct original values, but can make self-consistent values that reproduce
+            // the known distal targets
+            primaryAlignmentClippedAtStart = forwardStrand;
+            primaryAlignmentForwardStrand = forwardStrand;
+            saMappings = new ArrayList<>();
+            if(distalTargets.isEmpty()) {
+                tagSA = null;
+            } else {
+                for(final StrandedInterval distalTarget : distalTargets) {
+                    final String contigName = readMetadata.getContigName(distalTarget.getInterval().getContig());
```

Comment:
In the SV pipeline contig is stored as an integer code (to save memory). Since I need to occasionally use one to look the other up, I wanted to be clear which is which.